### DISCUSSION TITLE
Running Hardhat Network as a daemon

### DISCUSSION THREAD

**Discussion by JasperAlexander** [NomicFoundation/hardhat#2930] [Troubleshooting] [answered]
How to spin up an instance of Hardhat Network as a daemon?

**Comment by fvictorio** [comment, marked as answer]
I guess you mean running Hardhat Network in the background with a "detached" kind of flag (like the `--detach` flag in docker), right? If that's what you're asking, we don't have anything built-in for that.

Two things that come to mind are creating a Docker image with a running Hardhat network, or using something like [`pm2`](https://pm2.keymetrics.io/). There's also the simple option of just using `nohup`.

**Reply by JasperAlexander** [reply]
Yes that's what I mean. When are you going to built that? It seems to be quite easy using the [Forever package](https://www.npmjs.com/package/forever).

**Reply by fvictorio** [reply]
We don't have that on our roadmap. Also, this _might_ be something that can be done in user-land with a plugin, although I'm not 100% sure about that.

**Reply by AlphaDio** [reply]
This is really not great for development, yes Docker can be used, but if we want to automate our scripts and tasks through the Dockerfile, we should be able to start the Hardhat node FIRST in detach mode, and then do the scripts and tasks commands.

This makes Hardhat entirely useless as a local-ethereum blockchain in Docker, which is weird when Ganache offers the option.

**Reply by fvictorio** [reply]
@AlphaDio I agree that it would be useful, I was the one that opened [that suggestion in the Ganache repo](https://github.com/trufflesuite/ganache/issues/1001) :stuck_out_tongue:

We don't seem to have a feature request about starting the `node` task in detached mode though. Feel free to open one!
